acoustid/future/api/app.py

```
@@ -0,0 +1,90 @@
+async def handle_submission(request: Request) -> MsgspecResponse:
+    body = await request.body()
+    req = msgspec.json.decode(body, type=SubmissionRequest)
+
+    for submission in req.submissions:
+        try:
+            fp = await asyncio.to_thread(
+                decode_legacy_fingerprint,
+                submission.fingerprint,
+            )
+        except Exception as e:
+            raise ValidationError("Invalid fingerprint") from e
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)